plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeSink.java

```diff
@@ -83,10 +84,10 @@ public void add(int partitionId, Slice data)
         ExchangeStorageWriter writer = writers.computeIfAbsent(partitionId, this::createWriter);
 
         synchronized (writers.get(partitionId)) {
-            SliceOutput buffer = buffers.computeIfAbsent(partitionId, ignored -> Slices.allocate(exchangeStorage.getWriteBufferSizeInBytes()).getOutput());
+            dataBuffers.putIfAbsent(partitionId, Slices.allocate(exchangeStorage.getWriteBufferSizeInBytes()).getOutput());
```

Comment:
It should be either `put` (and maybe assert that there was no previous assignment) or `computIfAbsent`.